### DISCUSSION TITLE
Question about manipulation of documents in doctr

### DISCUSSION THREAD

**Discussion by Raul220** [mindee/doctr#1194] [Q&A]
Hello!
I want to know if when I use the doctr models on a document, as the following example:

model = ocr_predictor(det_arch='db_resnet50', reco_arch='crnn_vgg16_bn', pretrained=True)
pdf_doc = DocumentFile.from_pdf("path/to/your/doc.pdf")
result = predictor(doc)

and the models db_resnet50 and crnn_vgg16_bn are already in cache, does the predictor or from_pdf functions performs the processing locally?
Is the pdf document uploaded to the internet?

Thanks

**Comment by felixT2K** [comment]
Hey @Raul220 👋🏼 ,

everything runs on your machine. No files are uploaded anywhere. :)

All downoaded files (models & datasets) can be found in `~/.cache/doctr` you can also specify a custom location by setting the `DOCTR_CACHE_DIR` environment variable :)

ref.: `str(os.environ.get("DOCTR_CACHE_DIR", os.path.join(os.path.expanduser("~"), ".cache", "doctr")))`
